project/AkkaBuild.scala

```diff
@@ -81,97 +81,85 @@ object AkkaBuild {
+      mavenLocalResolverSettings ++
+      docLintingSettings ++ // compilerSettings had: javacOptions in doc ++= Seq(),
+      CrossJava.crossJavaSettings ++
+      Protobuf.settings ++
+      resolverSettings ++
+      Seq[Setting[_]](
+        crossVersion := CrossVersion.binary,
+
+        scalacOptions := CommonOptions.commonScalacOptions,
+
+        scalacOptions in Compile ++= CommonOptions.javaTarget(
+            scalaBinaryVersion.value, CrossJava.Keys.fullJavaHomes.value("8") + "/jre/lib/rt.jar"),
+
+        javacOptions := CommonOptions.commonJavacOptions,
+
+        javacOptions in compile ++= CommonOptions.javaSource(
+          CrossJava.Keys.fullJavaHomes.value("8") + "/jre/lib/rt.jar"),
+
+        // Adds a `src/main/scala-2.13+` source directory for Scala 2.13 and newer
+        // and a `src/main/scala-2.13-` source directory for Scala version older than 2.13
+        unmanagedSourceDirectories in Compile += {
+          val sourceDir = (sourceDirectory in Compile).value
+          sourceDir / (if (Versions.isScalaMinor(">=", 13, scalaVersion.value)) "scala-2.13+" else "scala-2.13-")
+        },
+
+        ivyLoggingLevel in ThisBuild := UpdateLogging.Quiet,
+
+        licenses := Seq(("Apache License, Version 2.0", url("http://www.apache.org/licenses/LICENSE-2.0"))),
+
+        homepage := Some(url("http://akka.io/")),
+
+        apiURL := Some(url(s"http://doc.akka.io/api/akka/${version.value}")),
+
+        initialCommands :=
+          """|import language.postfixOps
+           |import akka.actor._
+           |import ActorDSL._
+           |import scala.concurrent._
+           |import com.typesafe.config.ConfigFactory
+           |import scala.concurrent.duration._
+           |import akka.util.Timeout
+           |var config = ConfigFactory.parseString("akka.stdout-loglevel=INFO,akka.loglevel=DEBUG,pinned{type=PinnedDispatcher,executor=thread-pool-executor,throughput=1000}")
+           |var remoteConfig = ConfigFactory.parseString("akka.remote.netty{port=0,use-dispatcher-for-io=akka.actor.default-dispatcher,execution-pool-size=0},akka.actor.provider=remote").withFallback(config)
+           |var system: ActorSystem = null
+           |implicit def _system = system
+           |def startSystem(remoting: Boolean = false) { system = ActorSystem("repl", if(remoting) remoteConfig else config); println("don’t forget to system.terminate()!") }
+           |implicit def ec = system.dispatcher
+           |implicit val timeout = Timeout(5 seconds)
+           |""".stripMargin
+      )
+
+  val testSettings =
+    TestExtras.Filter.settings ++ Seq(
       fork in Test := true,
 
+      scalacOptions in Test := (scalacOptions in Test).value.filterNot(opt ⇒
+        opt == "-Xlog-reflective-calls" || opt.contains("genjavadoc")) ++ Seq("-Ywarn-unused"),
+
+      javacOptions in test ++= CommonOptions.commonJavacOptions ++ CommonOptions.javaSource(
+        CrossJava.Keys.fullJavaHomes.value("8") + "/jre/lib/rt.jar"),
```

Comment:
Ideally we can get this as one, e.g. from CrossJava, vs repeated but I'd need to set it up as a setting or task.